Feature: downsampled pdf

lsh-0 opened the pull request:
big update to strip-coverletter

* pdfsam-console has been replaced by sejda-console
* we now squash the pdf by 'downsampling' the images to 320dpi with a low-ish compression ratio. 
* we now have tests and CI
* we now have a Docker container that works
* a bug that allowed errors to go unlogged has been found and fixed

merge order:
* elife-bot-formula: https://github.com/elifesciences/elife-bot-formula/pull/13
* this

Commit:
upgraded pdfsam-console commands to sejda-console commands

Commit:
added initial downsample script

Commit:
downsample script now requires two args, infile and outfile

Commit:
integrated downsample script into strip-coverletter script

Commit:
updated dockerfile.
dockerfile now uses Arch rather than Ubuntu.

Commit:
merged in sejda branch

Commit:
fixed bug indockerfile

Commit:
added script that unzips the merge file from the zip
this is useful for running tests

Commit:
Dockerfile shifted to ./build and switched to ubuntu
tried to avoid pulling in the WM on Arch and Ubuntu but ultimately failed. at least with ubuntu it's just x11-common and not x itself

Commit:
replaced mentions of 'pdfsam-console' with 'sejda-console'

Commit:
cleaned up dockerfile. included a convenience script to build image

Commit:
script no longer jumps into temp directory
fixed up some messages

Commit:
log file is created as soon as possible

Commit:
dockerfile now suppresses git warnings when specific revision is built

Commit:
improved test script
expects a directory called bucket-pdf with the pdf files extracted from poa zips (see unzip-merged.sh)

Commit:
added quick script that emits csv rows of the original, decap and squashed file sizes in bytes

Commit:
added .gitignore. added quick script to download sejda-console

Commit:
added download-pdf.sh script.
this will download all of the zipped+merged poa pdf files from s3.

Commit:
strip-coverletter.sh now prefers decap to squashed if squashed is larger
this very occasionally happens at smaller file sizes when images are re-compressed at a higher dpi with less compression

Commit:
test.sh tweaks

Commit:
renamed test.sh to test-all.sh
this can be a very long test

Commit:
added potential alternative to strip command when first one fails

Commit:
new test.sh script

giorgiosironi commented on build/Dockerfile:
```diff
@@ -0,0 +1,22 @@
+RUN unzip -q sejda-console-3.2.38-bin.zip && ln -s sejda-console-3.2.38 sejda
+ENV PATH="/opt/strip-coverletter:/opt/strip-coverletter/sejda/bin:${PATH}"
+
+RUN git pull && git checkout --quiet a514bd43512b1a133668008cb8e698b0e408a18e
```
is missing a `CMD` statement to run a default script?

lsh-0 replied:
Dockerfile isn't finished yet unfortunately. Even if it were finished in this PR, I would still need to investigate how the bot is calling the script and update it. 

I got to the point of researching the different types of mounts Docker supports so that I could pass a file in and spit one out, but calling the Docker container [was taking much longer than calling the script directly](https://github.com/docker/for-mac/issues/1183#issuecomment-345899718 ) so I've just put it aside for now.

Commit:
Moved Dockerfile to top level to make COPY work

https://docs.docker.com/engine/reference/builder/#copy would require to specify a custom context directory

Commit:
Use files from current revision into container

Commit:
Added default script to run in container

Commit:
Showing input after bad arguments detection

Commit:
Added standard script for building image

Commit:
Add project tests

Commit:
Sample (failing) fixtures for project tests

Commit:
Basic pipeline building image and running tests with it

Commit:
Cannot run in interactive mode inside Jenkins pipelines

Commit:
replaced build_image with build/build-image

Commit:
project_tests.sh no longer expects caller to be root

Commit:
added .dockerignore file now that docker is being built in working dir

Commit:
added dummy pdf fixture.
added exit trap to script so any errors can be displayed.
updated the Dockerfile so copying new files doesn't mean reinstalling sejda.

Commit:
updated reference to old build_image script

lsh-0 committed:
Merge pull request #4 from elifesciences/pipeline

Pipeline

Commit:
removed old download-unpack.sh script

Commit:
download-sejda script now removes download afterwards

Commit:
improved ignore sejda pattern in .gitignore

Commit:
strip-coverletter now downloads sejda if it's not detected

Commit:
added download-test-fixtures.sh script
downloads a set of pdf files from S3 that the ./test.sh script can work on

Commit:
strip-coverletter now returns '2' if a coverletter cannot be found

Commit:
project_tests is now example-usage

Commit:
tweaked the test flags

Commit:
any existing sejda-console symlink is overwritten when downloading

giorgiosironi commented:
The build integrated here seems to work, creates the container and runs it correctly over the dummy file.

lsh-0 commented:
elife-bot-formula update has finally gone through. I'll merge this in tomorrow morning so I can debug anything after I merge.

It's not attached to elife-bot so I'll need to run highstate manually once merged and ensure it can be called without issue from the machine itself.

including @gnott as well
